src/components/TodoAppHeader/TodoAppHeader.tsx

```diff
@@ -0,0 +1,54 @@
+import React, {
+  useState,
+  useEffect,
+  useRef,
+} from 'react';
+
+type Props = {
+  addNewTodo: (title: string) => void;
+  isAdding: boolean;
+};
+
+export const TodoAppHeader: React.FC<Props> = ({
```

Comment:
Why don't you wrap all components with `React.memo`  ?

Reply:
As Ihor told us it is a bad practice to wrap everything in React.memo, as sometimes render happens quicker than checking whether we need or not to rerender component

Reply:
So what is the condition to use it then? Never or just when do you want it?

Reply:
When you know exactly that the most of the time some component will rerender in vain because of its parent, but makes sense to wrap Header